lib/publiccloud/ec2.pm

```diff
@@ -114,14 +114,40 @@ sub upload_img {
 
     die("Create key-pair failed") unless ($self->create_keypair($self->prefix . time, 'QA_SSH_KEY.pem'));
 
+    # AMI of image to use for helper VM to create/build the image on CSP.
+    my $helper_ami_id = get_var('PUBLIC_CLOUD_EC2_UPLOAD_AMI');
+
+    # in case AMI for helper VM is not provided in job settings fallback to predefined hash
+    unless (defined($helper_ami_id)) {
+
+        # AMI is region specific also we need to use different AMI's for on-demand/BYOS uploads
+        my $ami_id_hash = {
```

Comment:
Only remark: Is the ami of our helper VM stable enough to justify putting them as hard-coded within our tests?

Reply:
Take a look to the history of https://github.com/os-autoinst/os-autoinst-distri-opensuse/commits/master/data/publiccloud/ec2utils.conf it is quite stable.

@asmorodskyi had also the idea to use pint to get the latest image dynamically, but I voted against it, as each dynamic stuff makes the test more unstable (from my point of view).

Reply:
> Only remark: Is the ami of our helper VM stable enough to justify putting them as hard-coded within our tests?

1. I don't think that "hard-coded" is right term here - we still have `PUBLIC_CLOUD_EC2_UPLOAD_AMI` which you can put in your test suite / job group settings and this hash will even not be created :) 
2. This PR does not introduce "hard-coded AMI's" we already having the same "hard-coded AMI's "  just now they are in ec2utils.conf with a lot of not used garbage ( `accounts-*` config sections , `aki_*` variables in each regions etc. ) . With this PR I wipe out all this trash and leave only thing which we were really using **helper VM AMI** and put it in more appropriate place ( IMHO at least ).

Reply:
Well, those are still hard-coded variables, although they are not intended to be used as primary resource - But given the alternatives I find this here a valid approach.
I wanted to make sure that this is not a place that we need to constantly update and I got my answer from @cfconrad.

All good to go IMHO, and a nice solution for this problem. Good work Anton! :slightly_smiling_face: